getblocklab/block-lab
# Add a lightweight e2e test setup and tests

#### Changes
Adds an e2e test that runs in a headless browser and:
1. Activates Block Lab
2. Creates a new Block Lab block with a Text field
3. Create a new post
4. Adds that blocks to the post
5. Enters text in the text field
6. Clicks away from the block so the `<ServerSideRender>` displays
7. Verifies that the blocks looks good, using the PHP template that this PR adds. This tests that `block_value()` and `block_field()` have the correct output.

This uses the e2e test scripts from Gutenberg, minimizing the amount of code in this PR.

The test is simple because e2e tests can be really flaky. 

This should make it easier to merge `dependabot` PRs, knowing that the plugin works at a high level. 

For fine-grained JS tests, the [component tests](https://github.com/getblocklab/block-lab/tree/ad4981a3351c6136cc4d55b4cb2e21a182a54a96/js/blocks/components/test) are better, and have been reliable so far.

#### Testing instructions
```bash
npm install # might be necessary to remove node_modules first
npm run build
npm run env start
npm run test:e2e # To see the test running, do npm run test:e2e:interactive
npm run env stop # Do this whenever, but you probably don't want the container running all of the time.
```
Because this updates most of the packages in `package.json`, this could use full regression testing.
